kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -455,6 +466,40 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
+    /// Check for valid unchecked shift distance.
+    /// Shifts on an integer of type T are UB if shift distance < 0 or >= T::BITS.
+    fn check_unchecked_shift_distance(
+        &mut self,
+        value: &Operand<'tcx>,
+        distance: &Operand<'tcx>,
+        loc: Location,
+    ) -> Stmt {
+        let value_expr = self.codegen_operand(value);
+        let distance_expr = self.codegen_operand(distance);
+        let value_width = value_expr.typ().sizeof_in_bits(&self.symbol_table);
+        let value_width_expr = Expr::int_constant(value_width, distance_expr.typ().clone());
+
+        let excessive_distance_check = self.codegen_assert_assume(
+            distance_expr.clone().lt(value_width_expr),
+            PropertyClass::ArithmeticOverflow,
+            "attempt to shift by excessive shift distance",
+            loc,
+        );
+
+        // The first argument to the shift intrinsics determines the types of both arguments
```

Comment:
Gotcha. I added it because I thought it might be confusing that the signedness of `value_expr` is checked but then `distance_expr` is actually used. Now that I think about it again, should it just be checking if `distance_expr` is signed instead?

Reply:
Yeah, I think that would make more sense. Thanks!